actors/common/src/main/java/org/sunbird/learner/actors/otp/OTPActor.java

```diff
@@ -122,11 +120,32 @@ private void verifyOTP(Request request) {
       ProjectCommonException.throwClientErrorException(ResponseCode.errorInvalidOTP);
     }
 
+    if (otpInRequest.equals(otpInDB)) {
+      otpService.deleteOtp(type, key);
+    } else {
+      int remainingCount = getRemainingAttemptedCount(otpDetails);

```

Comment:
This complete else code can be moved to separate private method , as handleMisMatchOtp or some thing.